iothub/device/src/Transport/Mqtt/MqttLogger.cs

```diff
@@ -21,7 +22,7 @@ public void Publish(MqttNetLogLevel logLevel, string source, string message, obj
             string formattedMessage = message;
             if (parameters != null && parameters.Length > 0)
             {
-                formattedMessage = string.Format(message, parameters);
+                formattedMessage = string.Format(CultureInfo.CurrentCulture, message, parameters);
```

Comment:
I can see the point that a log statement should be formatted based on current culture; however, I wanted to get this confirmed from what the track 2 SDKs do.
Turns out, they format based on InvariantCulture, possibly to preserve the logging behavior across regions.

I'd recommend that we do the same.

https://github.com/Azure/azure-sdk-for-net/blob/e59553b4fb61b37c9c13681b4b117cc58de17e6d/sdk/core/Azure.Core/src/Diagnostics/AzureEventSourceListener.cs#L96

Reply:
Makes sense, updated